Add display failed tests option to slack notifications on job configuration

derinbay opened the pull request:
I've added an option as "Include Failed Tests" to slack notifications on job configuration screen to display failed tests.

PR#166 also adds the failed tests but I think there should be an option for user to choose on configurations. Also there was a conflict on that PR and it was very old, so I've opened this one.
<img width="580" alt="screen shot 2016-12-23 at 01 40 35" src="https://cloud.githubusercontent.com/assets/2565698/21442735/e86e0948-c8b0-11e6-9296-7b79800cbe68.png">

samrocketman commented:
Please rebase on master and resolve merge conflicts.

derinbay commented:
@samrocketman rebased and resolved conflicts.

samrocketman commented:
@derinbay if you've rebased, then why are there so many commits?

derinbay commented:
@samrocketman I guess I messed up with the rebase and made a merge from master instead, sorry about that.. Should I revert?

samrocketman commented:
@derinbay Do not revert.  Cherry pick the merge commit onto a new branch based on `upstream/master`.

Note: I haven't tested the following method.  I will test it tomorrow to be sure the commands are correct.

```bash
#from your own repository
git remote add upstream https://github.com/jenkinsci/slack-plugin
git fetch upstream
git checkout failed-tests-report
#rename the branch
git branch -m old-failed-tests-report
#create a new branch
git checkout upstream/master -b failed-tests-report
#get all of the changes as a single commit
git cherry-pick -m 2 old-failed-tests-report
#reuse metadata from your original commit
git commit --amend --reuse-message=1c86334fd262bd5a3caa82aeab6faab74948e2e9
```

Now you should be one commit ahead of upstream master.

```bash
#status should say you're one commit ahead of upstream/master
git status
#inspect your commit to ensure it is correct
git show HEAD
```

If everything looks okay to you then force push it to overwrite your branch.  **If you're unsure, then feel free to give me write access to your repository and I can fix it for you.**

derinbay commented:
@samrocketman Made some stuff but not sure if it's ok. First it was fine but then I revert back my last commit, that commit resolves a duplication which is not exist anymore, probably it was happened during the merge yesterday, so it was unnecessary. I've added you as collaborator to my repo. I'll be glad if you check. Thanks

samrocketman commented:
@derinbay I think you mistakenly messed up your branch.  The PR shows `0` files changed.

derinbay commented:
@samrocketman Ok I found the codes and saw that #166 was almost same development with this one. I've added the differences to this commit, that shows the duration of the failed test and fixed a lower case letter (I guess a typo)

samrocketman commented:
Looks good.  I'm going to test it before merging.  Thanks for your contribution.

samrocketman commented:
Feel free to subscribe to #296 for updates related to Slack Plugin 2.2 release.

blakecmartin commented:
Is it possible to send results in a pipeline using _slackSend_?
